tests/test_query.py

```diff
@@ -70,48 +76,69 @@ async def mock_get(sex):
 
 
 @pytest.mark.parametrize("valid_sex", ["male", "female", "other"])
-def test_get_input(test_app, valid_sex, monkeypatch):
+def test_get_sex(test_data, test_app, valid_sex, monkeypatch):
     """Given a valid sex string, returns a 200 status code and a non-empty list of results."""
-    test_data = [
-        {
-            "number_session": "test1",
-            "modality": "test1",
-            "subject": "test1",
-            "sub_id": "test1",
-            "sex": "test1",
-            "diagnosis": "test1",
-            "dataset_name": "test1",
-            "dataset": "test1",
-            "age": "test1",
-        },
-        {
-            "number_session": "test2",
-            "modality": "test2",
-            "subject": "test2",
-            "sub_id": "test2",
-            "sex": "test2",
-            "diagnosis": "test2",
-            "dataset_name": "test2",
-            "dataset": "test2",
-            "age": "test2",
-        },
-    ]
 
-    async def mock_get(sex):
+    async def mock_get(age_min, age_max, sex):
         return test_data
 
     monkeypatch.setattr(crud, "get", mock_get)
     response = test_app.get(f"/query/?sex={valid_sex}")
-    assert response.json() != []
     assert response.status_code == 200
+    assert response.json() != []
 
 
-def test_get_invalid_input(test_app, monkeypatch):
+def test_get_invalid_sex(test_app, monkeypatch):
     """Given an invalid sex string (i.e., anything other than ["male", "female", None]), returns a 422 status code."""
 
-    async def mock_get(sex):
+    async def mock_get(age_min, age_max, sex):
         return None
 
     monkeypatch.setattr(crud, "get", mock_get)
     response = test_app.get("/query/?sex=apple")
     assert response.status_code == 422
+
+
+@pytest.mark.parametrize(
+    "age_min_keyval, age_max_keyval",
+    [
+        ("age_min=30.5", "age_max=60"),
+        ("age_min=20.75", None),
+        (None, "age_max=50"),
```

Comment:
Case in point: after reading this test I went to look at the pydantic model and then scratched my head for 30 minutes because I was somehow primed to expect `None` as a valid user supplied value for `age_min` which I now understand is not the case. I do still not understand why you can provide `None` in the query string at all without the whole thing becoming invalid - but that's probably just a limitation of my knowledge of pydantic magic?